src/ASM/ASMs3Dmx.C

```diff
@@ -211,21 +222,22 @@ bool ASMs3Dmx::generateFEMTopology ()
     m_basis = ASMmxBase::establishBases(svol, ASMmxBase::Type);
 
     // we need to project on something that is not one of our bases
-    if (ASMmxBase::Type == ASMmxBase::REDUCED_CONT_RAISE_BASIS1 ||
-        ASMmxBase::Type == ASMmxBase::REDUCED_CONT_RAISE_BASIS2 ||
-        ASMmxBase::Type == ASMmxBase::DIV_COMPATIBLE)
-      projB = ASMmxBase::raiseBasis(svol);
-    else if (ASMmxBase::Type == ASMmxBase::SUBGRID) {
-      projB = m_basis.front()->clone();
+    if (!projB)
+      if (ASMmxBase::Type == ASMmxBase::REDUCED_CONT_RAISE_BASIS1 ||
+          ASMmxBase::Type == ASMmxBase::REDUCED_CONT_RAISE_BASIS2 ||
+          ASMmxBase::Type == ASMmxBase::DIV_COMPATIBLE)
+        projB = ASMmxBase::raiseBasis(svol);
+      else if (ASMmxBase::Type == ASMmxBase::SUBGRID)
+        projB = m_basis.front().get();
+      else // FULL_CONT_RAISE_BASISx
+        projB = m_basis[2-geoBasis].get();
```

Comment:
Same question on the added if as in the 2D case, and if yes, add the missing braces.

Reply:
same answer.